core/src/actors/sessions_manager.rs

```diff
@@ -127,34 +136,8 @@ impl Actor for SessionsManager {
     fn started(&mut self, ctx: &mut Self::Context) {
         debug!("Sessions Manager actor has been started!");
 
-        // Get address to launch the server
-        let config_manager_addr = System::current().registry().get::<ConfigManager>();
-
-        // Start chain of actions
-        config_manager_addr
-            // Send GetConfig message to config manager actor
-            // This returns a Request Future, representing an asynchronous message sending process
-            .send(GetConfig)
-            // Convert a normal future into an ActorFuture
-            .into_actor(self)
-            // Process the response from the config manager
-            // This returns a FutureResult containing the socket address if present
-            .then(|res, _act, _ctx| {
-                // Process the response from config manager
-                process_get_config_response(res)
-            })
-            // Process the received config
-            // This returns a FutureResult containing a success or error
-            .and_then(|config, act, _ctx| {
-                act.sessions
-                    .set_server_address(config.connections.server_addr);
-                act.sessions.set_limits(
-                    config.connections.inbound_limit,
-                    config.connections.outbound_limit,
-                );
-                actix::fut::ok(())
-            })
-            .wait(ctx);
+        // Send message to config manager and process its response
+        send_get_config_request(self, ctx, SessionsManager::process_config);
```

Comment:
You can use a closure instead of a function here, to avoid creating a new function:

```rust
// Send message to config manager and process its response
send_get_config_request(self, ctx, |s, _ctx, config| {
    s.sessions
        .set_server_address(config.connections.server_addr);
    s.sessions.set_limits(
        config.connections.inbound_limit,
        config.connections.outbound_limit,
    )
});
```